opensfm/src/sfm/src/tracks_helpers.cc

```diff
@@ -60,4 +68,154 @@ void RemoveConnections(map::TracksManager& manager, const map::ShotId& shot_id,
     manager.RemoveObservation(shot_id, connection);
   }
 }
+
+int FilterBadlyConditionedPoints(map::Map& map,
+                                double min_angle_deg,
+                                double min_abs_det) {
+  std::vector<map::LandmarkId> to_remove;
+
+  // Cull based on stats on covariance's condition number
+  constexpr double kMaxCond = 1000;
+  constexpr double kSigmaMultiplier = 1.0;
+  std::vector<std::pair<map::LandmarkId, double>> landmark_conds;
+  landmark_conds.reserve(map.GetLandmarks().size());
+
+  // Iterate over a snapshot of landmarks to decide removals.
+  int removed_triangulation = 0;
+  for (const auto& kv : map.GetLandmarks()) {
+    const map::LandmarkId& lm_id = kv.first;
+    const map::Landmark& lm = kv.second;
+
+    const auto& observations = lm.GetObservations();
+
+    std::vector<geometry::Camera> cameras;
+    cameras.reserve(observations.size());
+    std::vector<geometry::Pose> poses;
+    poses.reserve(observations.size());
+    std::vector<Vec2d> obs_vec;
+    obs_vec.reserve(observations.size());
+
+    // Build camera/pose/observation vectors from observations.
+    for (const auto& obs_kv : observations) {
+      map::Shot* shot_ptr = obs_kv.first;
+      if (!shot_ptr) {
+        continue;
+      }
+
+      const geometry::Pose* pose_ptr = shot_ptr->GetPose();
+      const geometry::Camera* cam_ptr = shot_ptr->GetCamera();
+      if (!pose_ptr || !cam_ptr) {
+        continue;
+      }
+
+      // Get the shot's Observation for this landmark.
+      map::Observation* obs_ptr = shot_ptr->GetLandmarkObservation(
+          const_cast<map::Landmark*>(&lm));
+      if (!obs_ptr) {
+        continue;
+      }
+
+      cameras.push_back(*cam_ptr);
+      poses.push_back(*pose_ptr);
+      obs_vec.push_back(obs_ptr->point);
+    }
+
+    // First simple check based on angle between all pairs of raysmap
```

Comment:
We do this check before triangulating the points. Is the idea that after bundle that could have changed?